### PR TITLE
ZWEB-7348 Adapt to Modern UI-Router

### PR DESCRIPTION
**Pull request by Jacob-Friesen**
Right now there does not seem to be unit tests for this repository, so it is best to verify with the main application.

### CODE REVIEW

File: dist/zerv-ng-security.js [hunk 1]
```diff
@@ -63,21 +63,9 @@
...
-      //         // return toState.redirectTo.indexOf(deniedState.resource.locator) === 0;
-      //     });
-      //     if (denied) {
-      //         event.preventDefault();
-      //     }
-      //     //$state.go(toState.redirectTo, toParams);
-      // }
-    });
-    $rootScope.$on('$stateChangeSuccess', function (event, toState, toParams, fromState, fromParams) {
+    $transitions.onSuccess({}, function () {
```
Comment (etelmon) [inline dist/zerv-ng-security.js:68]: changes should have unit tests. we need to start at some point

File: src/resources/ui-state.resource.js [hunk 2]
```diff
@@ -1,32 +1,15 @@
...
-
-        $rootScope.$on('$stateChangeSuccess',
-            function(event, toState, toParams, fromState, fromParams) {
-                const denied = _.find(deniedStates, function(deniedState) {
-                    return $state.includes(deniedState.stateName);
-                });
-                if (denied) {
-                    redirect(denied);
-                }
+        $transitions.onSuccess({}, () => {
```
Comment (etelmon) [inline src/resources/ui-state.resource.js:5]: same here, let's start adding unit tests

**Review by etelmon — CHANGES_REQUESTED** [review]

File: dist/zerv-ng-security.js (hunk 1 above)
Comment (etelmon) [reply dist/zerv-ng-security.js:68]: The generated lib file was modified instead of the src file.

File: dist/zerv-ng-security.js [hunk 3]
```diff
@@ -63,21 +63,9 @@
 "use strict";
```
Comment (etelmon) [inline dist/zerv-ng-security.js:63]: package  should be increased to "version": "0.0.5",

**Review by etelmon — CHANGES_REQUESTED** [review]

**Comment by etelmon** [comment]
Besides the comments above, The new $transition implementation works as expected.

File: src/resources/ui-state.resource.js (hunk 2 above)
Comment (Jacob-Friesen) [reply src/resources/ui-state.resource.js:5]: Added

**Comment by etelmon** [comment]
Tag is not created yet for "version": "0.0.5",

**Comment by Jacob-Friesen** [comment]
The tag was created here after merge: https://github.com/z-open/zerv-ng-security/releases/tag/0.0.5

**Pull request merged**